### PR TITLE
KAZOO-4588: exctract userid properly in cb_apps_util:is_authorized

### PR DESCRIPTION
**Pull request by icehess**

### CODE REVIEW

File: applications/crossbar/src/cb_apps_util.erl [hunk 1]
```diff
@@ -63,11 +63,12 @@ is_authorized(AccountId, UserId, AppId) ->
             AppJObj = wh_json:get_value(AppId, kz_apps_store:apps(Doc)),
             Allowed = wh_json:get_value(<<"allowed_users">>, AppJObj, <<"specific">>),
             Users = wh_json:get_value(<<"users">>, AppJObj, []),
-            case {Allowed, Users} of
+            UserIds = [wh_json:get_value(<<"id">>, U, U) || U <- Users],
```
Comment (fenollp) [inline applications/crossbar/src/cb_apps_util.erl]: Default value is the jobj `U`?

Comment (icehess) [inline applications/crossbar/src/cb_apps_util.erl]: @fenollp typo, I'll fix it.

File: applications/crossbar/src/cb_apps_util.erl [hunk 2]
```diff
@@ -63,11 +63,12 @@ is_authorized(AccountId, UserId, AppId) ->
...
             Allowed = wh_json:get_value(<<"allowed_users">>, AppJObj, <<"specific">>),
             Users = wh_json:get_value(<<"users">>, AppJObj, []),
-            case {Allowed, Users} of
+            UserIds = [wh_json:get_value(<<"id">>, U, []) || U <- Users],
+            case {Allowed, UserIds} of
                 {<<"all">>, _} -> 'true';
                 {<<"specific">>, []} -> 'false';
-                {<<"specific">>, Users} when is_list(Users) ->
-                    lists:member(UserId, Users);
+                {<<"specific">>, UserIds} when is_list(UserIds) ->
```
Comment (lazedo) [inline applications/crossbar/src/cb_apps_util.erl]: `when is_list(UserIds)` i believe this is not needed ?

File: applications/crossbar/src/cb_apps_util.erl [hunk 3]
```diff
@@ -63,11 +63,12 @@ is_authorized(AccountId, UserId, AppId) ->
             AppJObj = wh_json:get_value(AppId, kz_apps_store:apps(Doc)),
             Allowed = wh_json:get_value(<<"allowed_users">>, AppJObj, <<"specific">>),
             Users = wh_json:get_value(<<"users">>, AppJObj, []),
-            case {Allowed, Users} of
+            UserIds = [wh_json:get_value(<<"id">>, U, []) || U <- Users],
```
Comment (fenollp) [inline applications/crossbar/src/cb_apps_util.erl]: Why default to `""`?

Also, shouldn't all occurrences of that default value be removed from the list? Matching on `[]` and `lists:member` look like they could be harmed by `undefined` being where it shouldn't be

**Pull request merged**